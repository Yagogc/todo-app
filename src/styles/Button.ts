import styled, { css } from 'styled-components/macro'

const Button = styled.button`
  ${() => css`
    padding: 8px;
    background-color: transparent;
    border: 1px solid var(--secondary-bg-color);

    &:hover {
      cursor: pointer;
    }
  `}
`

export { Button }
