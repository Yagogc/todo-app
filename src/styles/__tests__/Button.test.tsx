import React from 'react'
import { render } from '@testing-library/react'
import { Button } from '../Button'

describe('Button', () => {
  it('has default styles', () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer', {
      modifier: ':hover',
    })
  })
})
