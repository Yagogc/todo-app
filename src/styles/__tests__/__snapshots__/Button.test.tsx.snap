// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has default styles 1`] = `
.c0 {
  padding: 8px;
  background-color: transparent;
  border: 1px solid var(--secondary-bg-color);
}

.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
/>
`;
